components/NavBar.tsx

```diff
@@ -1,28 +1,45 @@
+"use client";
 import Image from "next/image";
 import Link from "next/link";
-import { Lobster } from "next/font/google";
+import { Lobster, Inter } from "next/font/google";
+import { useSession } from "next-auth/react";
 
 const lobster = Lobster({ weight: "400", subsets: ["latin"] });
-import Icon from "./Icon";
+const inter = Inter({ weight: "400", subsets: ["latin"] });
+import { SessionProvider } from "next-auth/react";
 
 export default function NavBar() {
+  const { data: session, status } = useSession();
   return (
-    <header
-      className={`flex justify-between items-center border-b border-gray-800 p-4 ${lobster.className}`}
-    >
-      <Link href="/">
-        <h1 className="text-2xl">Collections</h1>
-      </Link>
+    <SessionProvider>
+      <header
+        className={`flex justify-between items-center border-b border-gray-800 p-4 ${lobster.className}`}
+      >
+        <Link href="/">
+          <h1 className="text-2xl">Collections</h1>
+        </Link>
 
-      <Link href="/user-profile">
-        <Image
-          src="/images/profile-picture.jpg"
-          width={32}
-          height={32}
-          alt="User profile picture"
-          className="rounded-full"
-        />
-      </Link>
-    </header>
+        {status === "authenticated" ? (
+          <div className="flex flex-row gap-3 items-center">
+            <Link href="/user-profile">
+              <Image
+                src="/images/profile-picture.jpg"
+                width={32}
+                height={32}
+                alt="User profile picture"
+                className="rounded-full"
+              />
+            </Link>
+            <Link href={"/api/auth/signout"}>
+              <button className={`${inter.className} `}>Sign Out</button>
+            </Link>
+          </div>
+        ) : (
+          <Link href="/api/auth/signin">
```

Comment:
I belive you'd want to use [NextAuth's signIn function here](https://next-auth.js.org/getting-started/client#signin)

Reply:
Wasn't sure what the difference was. Just read the Auth.js docs though and it seems like the signIn() and signOut() functions handle CSRF tokens for you. Seems that there is additional functionality that comes with these functions so makes sense why I need to use those instead of a Link component.